packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -310,9 +329,26 @@ abstract class ScrollView extends StatelessWidget {
         return buildViewport(context, offset, axisDirection, slivers);
       },
     );
-    return primary && scrollController != null
-      ? PrimaryScrollController.none(child: scrollable)
-      : scrollable;
+    final Widget scrollableResult = primary && scrollController != null
+        ? PrimaryScrollController.none(child: scrollable)
+        : scrollable;
+
+    if (keyboardDismissBehavior == ScrollViewKeyboardDismissBehavior.onDrag) {
+      return NotificationListener<ScrollUpdateNotification>(
+        child: scrollableResult,
+        onNotification: (ScrollUpdateNotification notification) {
+          final FocusScopeNode focusScope = FocusScope.of(context);
+          if (notification.dragDetails != null &&
+              (focusScope.focusedChild != null &&
```

Comment:
Why do you need this to write this differently from `focusScope.hasFocus`?  And if that's the case, then why wouldn't this work?:

```dart
if (notification.dragDetails != null) {
  focusScope.unfocus();
}
```

Basically, what's the reason for the extra conditions?

Reply:
`notifcation.dragDetails != null` is to make sure the scrolling is the result of a drag, these events get fired too when the scrollview is scrolled as the result of a keyboard showing.

`focusScope.focusedChild != null` is because I'm going to read from the focusedChild and I don't want to read from a null pointer.

`focusScope.focusedChild.hasFocus` makes sure I don't spam `focusScope.unfocus()` and that it only gets called once for all the scrolling events that get sent.

Reply:
Could I just used `focusScole.hasFocus` and avoid talking to the focusedChild?

Reply:
I switched it over to `focusScope.hasFocus`.  It wasn't like this because i found the other way to do it first and it worked.